pkg/snapshot/snapshotter/garbagecollector.go

```diff
@@ -153,7 +153,11 @@ func (ssr *Snapshotter) RunGarbageCollector(stopCh <-chan struct{}) {
 							continue
 						}
 						ssr.logger.Infof("GC: Deleting old full snapshot: %s %v", nextSnap.CreatedOn.UTC(), deleteSnap)
-						if err := ssr.store.Delete(*nextSnap); err != nil {
+						if err := ssr.store.Delete(*nextSnap); errors.Is(err, brtypes.ErrSnapshotDeleteFailDueToImmutability) {
```

Comment:
here `ssr.store.Delete(*nextSnap)` will eventually call `Delete()` from ABS package ... right ?
so, IMO you have already included the error check of `Immutability`, so why to rehandling it again here ?

```
	if _, err := blobClient.Delete(context.Background(), nil); bloberror.HasCode(err, bloberror.BlobImmutableDueToPolicy) {
		return fmt.Errorf("failed to delete blob %s due to immutability: %w, with provider error: %w", blobName, brtypes.ErrSnapshotDeleteFailDueToImmutability, err)
	}
```

Reply:
The API request to ABS does not populate the immutability duration when the immutability period is set only to the bucket. The request returns a `null` for that field.

Thus there is no information that can be used to fetch the duration till which the snapshot is immutable.

The only way to find this information is to attempt a delete, and if the delete request errors with the specific code that corresponds to immutability, we know the deletion fails because of immutability. When such an event happens, we simply ignore the snapshot from GC and move ahead.

Reply:
As discussed with @renormalize , we handled `immutableError` separatly because we don't want it to included in metrics of failed GC of snapshots of backup-restore as immutable backups can't be GC, so it's expected.